src/ert/scheduler/local_driver.py

```diff
@@ -64,17 +68,25 @@ async def _run(self, iens: int, executable: str, /, *args: str) -> None:
             # /bin/sh uses returncode 127 for FileNotFound, so copy that
             # behaviour.
             logger.error(f"Realization {iens} failed with {err}")
-            await self.event_queue.put(FinishedEvent(iens=iens, returncode=127))
+            await self._dispatch_finished_event(iens, 127)
             return
 
         await self.event_queue.put(StartedEvent(iens=iens))
+
+        returncode = 0
         try:
             returncode = await self._wait(proc)
             logger.debug(f"Realization {iens} finished with {returncode=}")
-            await self.event_queue.put(FinishedEvent(iens=iens, returncode=returncode))
         except asyncio.CancelledError:
             returncode = await self._kill(proc)
+        finally:
+            await self._dispatch_finished_event(iens, returncode)
```

Comment:
Maybe there should be another ?
```python
except Exception:
    returncode = signal.SIGTERM + SIGNAL_OFFSET
```

Reply:
If anything else than CancelledError occurs here, we should crash (for the moment at least)